### PR TITLE
Magic link add link back to login

### PR DESCRIPTION
**Pull request by jlopes90**

### CODE REVIEW

**Comment by jlopes90** [comment]
Because of "feat: add filter permission and group" the branch is develop. I forget a lot of time to create branch.

File: src/Language/es/Auth.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace CodeIgniter\Shield\Language\es;
```
Comment (kenjis) [inline src/Language/es/Auth.php:3]: This file is already in `develop`.
Please don't include changes other than this PR.

**Comment by kenjis** [comment]
This branch has conflicts that must be resolved.
Please rebase:
```console
$ git fetch upstream
$ git rebase upstream/develop
```
Resolve conflict, and `git add` the file.

```console
$ git rebase --continue
```

File: src/Language/de/Auth.php [hunk 2]
```diff
@@ -35,6 +35,7 @@
 
     // Login
     'login'              => 'Anmelden',
+    'backLogin'          => 'Zurück zur Anmeldung',
```
Comment (paulbalandan) [inline src/Language/de/Auth.php:38]: I think `backToLogin` as key is better than `backLogin`

**Comment by jlopes90** [comment]
imagem

I did something wrong, and it doesn't show upstream.

**Comment by jlopes90** [comment]
imagem

**Comment by kenjis** [comment]
Run `git remove -v`.

**Comment by jlopes90** [comment]
imagem

**Comment by kenjis** [comment]
Sorry, `git remote -v`.

**Comment by jlopes90** [comment]
imagem

**Comment by jlopes90** [comment]
I think I already solved it, I used `git remote add upstream https://github.com/codeigniter4/shield.git -f`

**Comment by datamweb** [comment]
@jlopes90 , I want you to write a **unittest** for this PR.
Writing a test for this PR is very simple.
I want you to get familiar with the test and this PR is the best place to start.
can help you:
https://codeigniter4.github.io/CodeIgniter4/testing/response.html

https://github.com/codeigniter4/shield/blob/c48cb81bd80762e0133d1ba53f50e16ab99c1f6e/tests/Collectors/AuthTest.php#L40-L54
You can write the test in `tests/Controllers/MagicLinkTest.php`.
eg:
```php 
 public function testHasLinkBackToLogin(): void
{
// test code
}
```
Please do your best, and let me know if you can't.

**Comment by jlopes90** [comment]
> https://github.com/codeigniter4/shield/pull/375#issuecomment-1214879964

Unfortunately, I tried to figure it out and I still don't understand how things work.

**Comment by datamweb** [comment]
> > [#375 (comment)](https://github.com/codeigniter4/shield/pull/375#issuecomment-1214879964)
> 
> Unfortunately, I tried to figure it out and I still don't understand how things work.

It's okay, you'll learn it eventually. In fact, this test is not important for this PR. Look [here](https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/README.md), it will help you a lot.

**Comment by datamweb** [comment]
@jlopes90, change it:
```php
 </form>
   <p class="text-center"><a href="<?= url_to('login') ?>"><?= lang('Auth.backToLogin') ?></a></p>
```

File: src/Views/magic_link_form.php [hunk 3]
```diff
@@ -9,37 +9,38 @@
...
 
             </form>
+
+            <p class="text-center"><a href="<?= url_to('login') ?>"><?= lang('Auth.backToLogin') ?></a></p>
         </div>
     </div>
 </div>
 
-<?= $this->endSection() ?>
+<?= $this->endSection() ?>
```
Comment (kenjis) [inline src/Views/magic_link_form.php:46]: Please add a line break.

File: src/Views/magic_link_form.php [hunk 4]
```diff
@@ -9,37 +9,38 @@
...
+            <?php endif ?>
 
             <form action="<?= url_to('magic-link') ?>" method="post">
                 <?= csrf_field() ?>
 
                 <!-- Email -->
                 <div class="mb-2">
-                    <input type="email" class="form-control" name="email" autocomplete="email" placeholder="<?= lang('Auth.email') ?>"
-                           value="<?= old('email', auth()->user()->email ?? null) ?>" required />
+                    <input type="email" class="form-control" name="email" autocomplete="email" placeholder="<?= lang('Auth.email') ?>" value="<?= old('email', auth()->user()->email ?? null) ?>" required />
```
Comment (kenjis) [inline src/Views/magic_link_form.php:32]: Please add a line break as before.

File: src/Views/magic_link_form.php (hunk 3 above)
Comment (jlopes90) [reply src/Views/magic_link_form.php:46]: Like this? It looks the same.

Comment (jlopes90) [reply src/Views/magic_link_form.php:46]: Original
```php
        </div>
    </div>
</div>

<?= $this->endSection() ?>
```

Change by @kenjis
```php
        </div>
    </div>
</div>


<?= $this->endSection() ?>
```

two line breaks?

Comment (kenjis) [reply src/Views/magic_link_form.php:46]: No, at the end of the file.

`<?= $this->endSection() ?>\n`

Comment (kenjis) [reply src/Views/magic_link_form.php:46]: @jlopes90 I recommend you use `.editorconfig` aware IDE.
See https://editorconfig.org/

**Comment by jlopes90** [comment]
oops, wrong duplicate commits

**Comment by kenjis** [comment]
I don't know why you committed the duplicate commits.

Please don't use `git merge` on the PR branch.
See the workflow: https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md

**Pull request closed without merging**